app/src/main/java/com/example/tasky/feature_agenda/data/worker/SyncWorker.kt

```diff
@@ -12,30 +12,34 @@ import com.example.tasky.feature_agenda.domain.util.AgendaItemType.EVENT
 import com.example.tasky.feature_agenda.domain.util.AgendaItemType.REMINDER
 import com.example.tasky.feature_agenda.domain.util.AgendaItemType.TASK
 import com.example.tasky.feature_agenda.domain.util.OperationType
-import com.squareup.moshi.Moshi
-import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.firstOrNull
 
 class SyncWorker(
     private val context: Context,
     private val workerParams: WorkerParameters,
-    private val moshi: Moshi,
     private val repositories: AgendaRepositories,
     private val db: AgendaDatabase
 ) : CoroutineWorker(context, workerParams) {
     override suspend fun doWork(): Result {
 
-        val syncItems = db.agendaDao.getItemsToBeSynced().first()
+        val syncItems = db.agendaDao.getItemsToBeSynced().firstOrNull()
```

Comment:
In which case would this be null? Isn't this just a list and if there are no items, it would just return an empty one?